### PR TITLE
Improve doc

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Update documentation structure and content

* **What is the current behavior?** (You can also link to an open issue here)
Compilation instructions and dependency management is not clear and spread across pages

* **What is the new behavior (if this is a feature change)?**
Simplify documentation to help new users understanding the build chain and use it.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: doc/basics/compilation.md [hunk 1]
```diff
@@ -23,25 +9,37 @@ The following platforms and tool chains have been tested and should work :
  * *Mac OSX* : gcc 10 or higher, Apple clang
  * *Linux* : gcc 8  or higher, clang
 
-## Continuous Integration:
- * *Linux (gcc10), Windows (MSVC 19.28.29335.0), and Mac OSX (AppleClang 12)* https://github.com/STORM-IRIT/Radium-Engine/actions
+See also our Continuous Integration system at https://github.com/STORM-IRIT/Radium-Engine/actions.
+
+Minimal requirements
+* OpenGL 3+ / GLSL 330+
```
Comment (MathiasPaulin) [inline doc/basics/compilation.md:15]:
```suggestion
* OpenGL 4.1+ / GLSL 410+
```

File: doc/basics/dependencies.md [hunk 2]
```diff
@@ -1,54 +1,93 @@
...
+* [Core] Eigen, OpenMesh
+* [Engine] glm, globjects, glbindings
+* [IO] Assimp
+* [Gui] Qt Core, Qt Widgets and Qt OpenGL v5.5+ (5.14 recommended)
+* [doc] Doxygen-awesome-css
+* stb_image
 
-- For release build :
+We developed a series of tools to fetch and compile easily these dependencies easily, except for
+Qt, which needs to installed and passed to cmake through the variable `Qt5_DIR` (see documentation at
```
Comment (MathiasPaulin) [inline doc/basics/dependencies.md:13]:
```suggestion
Qt, which needs to be installed and passed to cmake through the variable `Qt5_DIR` (see documentation at
```

**Review by MathiasPaulin — COMMENTED** [review]
Seems quite a good improvement in readability. I'll fetch and compile the doc to verify the new structure and content. 
BTW, perhaps you can add somewhere that we encourage users to have dedicated build directory and install directory (the later being manage automatically if the default install dir is used) for each build type (Release, Debug) and that compiling Radium in Debug mode needs to have the dependencies (at least assimp) compiled and installed in Debug mode.

**Comment by nmellado** [comment]
> Seems quite a good improvement in readability. I'll fetch and compile the doc to verify the new structure and content.
> BTW, perhaps you can add somewhere that we encourage users to have dedicated build directory and install directory (the later being manage automatically if the default install dir is used) for each build type (Release, Debug) and that compiling Radium in Debug mode needs to have the dependencies (at least assimp) compiled and installed in Debug mode.

Hum right, this is suggested at multiple places, but not clearly stated. I'll let you do you review run, and we will add that with your other suggestions (if any).

File: doc/basics/compilation.md [hunk 3]
```diff
@@ -118,52 +116,66 @@ All radium related cmake options (with their current values) can be printed with
...
 $ make install
 ~~~
 
+\note Running the `install` target is recommended as it will copy all the radium related library in the same place,
+generate the cmake packages and bundle applications with their dependencies (on macos and windows).
+
 Note that installation requires write access on the installation directory.
 
-## Building on Microsoft Windows with Visual Studio
+## Building note for Microsoft Windows with Visual Studio
```
Comment (dlyr) [inline doc/basics/compilation.md:155]:
```suggestion
## Integration with Visual Studio (Microsoft Windows)
```

File: doc/main.md [hunk 4]
```diff
@@ -37,5 +37,5 @@ See [this presentation](https://docs.google.com/presentation/d/12W2KXY7ctJXFIelm
 
 The documentation is structured as follow:
  - @ref basicsmanual : how to compile Radium and use it in your own project.
- - @ref conceptsmanual : high level concepts use in Radium Engine
  - @ref develmanual : general topics (e.g. coding conventions) and description of the concepts found in the API of Radium.
+- @ref conceptsmanual : high level concepts used in Radium Engine
```
Comment (dlyr) [inline doc/main.md:41]:
```suggestion
 - @ref conceptsmanual : high level concepts used in Radium Engine
```

File: doc/basics/dependencies.md [hunk 5]
```diff
@@ -1,54 +1,93 @@
...
+Radium relies on several external libraries to load files or to represent some data:
+* [Core] Eigen, OpenMesh
+* [Engine] glm, globjects, glbindings
+* [IO] Assimp
+* [Gui] Qt Core, Qt Widgets and Qt OpenGL v5.5+ (5.14 recommended)
+* [doc] Doxygen-awesome-css
+* stb_image
 
-- For release build :
+We developed a series of tools to fetch and compile easily these dependencies easily, except for
```
Comment (dlyr) [inline doc/basics/dependencies.md:12]:
```suggestion
We developed a series of tools to fetch and compile these dependencies easily, except for
```

**Review by dlyr — APPROVED** [review]

**Comment by dlyr** [comment]
Maybe clean the history before merge ? (or squash, but I think two commit may be better)

**Review by MathiasPaulin — APPROVED** [review]

**Comment by MathiasPaulin** [comment]
In the next days (or weeks), I'll propose a new section on "How to write your own plugin" so that even applications that do not support plugins can used the added functionalities (e.g. headless app). I gave an online course to master students, on the spirit of the "ExamplePluginWithLib" test folder, on how to do that and I just have to clean the used repo before submitting it as a PR.

**Comment by nmellado** [comment]
> In the next days (or weeks), I'll propose a new section on "How to write your own plugin" so that even applications that do not support plugins can used the added functionalities (e.g. headless app). I gave an online course to master students, on the spirit of the "ExamplePluginWithLib" test folder, on how to do that and I just have to clean the used repo before submitting it as a PR.

Btw, there is already a page called *how to write your own plugin* in the developer manual

**Comment by MathiasPaulin** [comment]
> Btw, there is already a page called _how to write your own plugin_ in the developer manual

Sure, there is one and it is helpful for developing plugins. 
But I think that this way of extending Radium by developing plugins is very limited as only BaseApplication derived apps will benefit from the functionalities. 
If we want also headless-app (or non Qt-based GUI apps) to benefit from these, we should prefer to develop a library exposing the functionality and interface this library with BaseApplication derived apps by developing a plugin while headless-app will directly link to the library.
This way of extending Radium was the aim of the demo "ExamplePluginWithLib" but, for now, this demo was not updated according to recent PR and does not compile anymore (it is not included in the CI nor in the tests).
There are also other advantages of extending Radium by a lib + plugin, we will discuss about this in a next steering committee.

**Pull request merged**